google/cloud/bigtable/data/_metrics/handlers/opentelemetry.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+from uuid import uuid4
+
+from google.cloud.bigtable import __version__ as bigtable_version
+from google.cloud.bigtable.data._metrics.handlers._base import MetricsHandler
+from google.cloud.bigtable.data._metrics.data_model import OperationType
+from google.cloud.bigtable.data._metrics.data_model import DEFAULT_CLUSTER_ID
+from google.cloud.bigtable.data._metrics.data_model import DEFAULT_ZONE
+from google.cloud.bigtable.data._metrics.data_model import ActiveOperationMetric
+from google.cloud.bigtable.data._metrics.data_model import CompletedAttemptMetric
+from google.cloud.bigtable.data._metrics.data_model import CompletedOperationMetric
+
+
+class _OpenTelemetryInstruments:
+    """
+    class that holds OpenTelelmetry instrument objects
+    """
+
+    def __init__(self, meter_provider=None):
+        if meter_provider is None:
+            # use global meter provider
+            from opentelemetry import metrics
+
+            meter_provider = metrics
+        # grab meter for this module
+        meter = meter_provider.get_meter("bigtable.googleapis.com")
+        # create instruments
+        self.operation_latencies = meter.create_histogram(
+            name="operation_latencies",
+            description="A distribution of the latency of each client method call, across all of it's RPC attempts",
+            unit="ms",
+        )
+        self.first_response_latencies = meter.create_histogram(
+            name="first_response_latencies",
+            description="A distribution of the latency of receiving the first row in a ReadRows operation.",
+            unit="ms",
+        )
+        self.attempt_latencies = meter.create_histogram(
+            name="attempt_latencies",
+            description="A distribution of the latency of each client RPC, tagged by operation name and the attempt status. Under normal circumstances, this will be identical to operation_latencies. However, when the client receives transient errors, operation_latencies will be the sum of all attempt_latencies and the exponential delays.",
+            unit="ms",
+        )
+        self.retry_count = meter.create_counter(
+            name="retry_count",
+            description="A count of additional RPCs sent after the initial attempt. Under normal circumstances, this will be 1.",
```

Comment:
Under normal circumstances, this value is empty. https://cloud.google.com/bigtable/docs/client-side-metrics-descriptions#retry-count